tests/phpunit/tests/admin/includesMedia.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ * @group media
+ * @group admin
+ */
+class Tests_Admin_includesMedia extends WP_UnitTestCase {
+	/** @var \WP_Post */
+	protected static $post;
+	/** @var \WP_Post */
+	protected static $different_post;
+
+	public static function wpSetUpBeforeClass( \WP_UnitTest_Factory $factory ) {
+		self::$post           = $factory->post->create_and_get();
+		self::$different_post = $factory->post->create_and_get();
+	}
+
+	public function wpTearDownAfterClass() {
+		self::$post           = null;
+		self::$different_post = null;
+	}
+
+	public function setUp() {
+		parent::setUp();
+
+		add_filter( 'pre_http_request', [ $this, 'internal_fake_download_url' ], 10, 3 );
+	}
+
+	public function tearDown() {
```

Comment:
This teardown is also not needed. The reason you need to add the filter in `setUp()` is because it runs before every test method and filters get reset between each test method so the filter will already be removed, and WP should clean up uploads itself, I believe.

Reply:
Check the teardown on the Unit Test `Tests_Post_Attachments` they do the same to clear the file system. Theme Dir they remove filter on tearDown.